lib/Data/Xls.hs

```diff
@@ -111,6 +162,27 @@ decodeXls file =
             count <- liftIO $ c_xls_wb_sheetcount pWB
             mapM_ (decodeOneWorkSheet file pWB) [0 .. count - 1]
 
+-- | A work-around via temporary files and 'decodeXlsIO', 
+-- since this lib is lacking a pure function to decode the contents 
+-- of an XLS file. 
+-- Due to Erik Rybakken. 
+decodeXlsByteString :: ByteString -> IO [[[Cell]]]
+decodeXlsByteString content = withSystemTempFile "decodeXlsByteString"
+    $ \filePath h -> do
+        hPut h content
+        decodeXlsIO filePath
+
+-- | Experimental: This function uses the @xls_open_buffer@ function of libxls.
+decodeXlsByteString' :: ByteString -> IO (Either XLSError [[[Cell]]])
```

Comment:
1. It would be better to have the same signature as decodeXlsByteString. The only difference between the two is exception vs either return. Both are IO, so can use exception.
2. We can say in the docs, that it does not use a temporary file, instead uses ...

Reply:
So you suggest to make `XLSError` an instance of `Exception` and use `throwIO` instead of explicitly returning `Either`?

Reply:
I created a [branch](https://github.com/olafklinke/xls/pull/1) in my fork for unifying the error types.